higherKindCore/src/main/scala/tofu/control/Bind.scala

```diff
@@ -0,0 +1,153 @@
+      leftFlatMap(fab)(a => left(f(a)), b => right(g(b)))
+  }
+
+  def rightBifunctor: Bifunctor[R] = new Bifunctor[R] {
+    def bimap[A, B, C, D](fab: R[A, B])(f: A => C, g: B => D): R[C, D] =
+      rightFlatMap(fab)(a => left(f(a)), b => right(g(b)))
+  }
+}
+
+trait TwinMonad[F[_, _]] extends BiMonad[F, F] with Bifunctor[F] { self =>
+  def pure[E, A](a: A): F[E, A]
+
+  def raise[E, A](e: E): F[E, A]
+
+  def foldWith[E, A, X, R](fa: F[E, A], h: E => F[X, R], f: A => F[X, R]): F[X, R]
+
+  final def foldWithC[E, A, X, R](fa: F[E, A])(h: E => F[X, R])(f: A => F[X, R]): F[X, R] =
+    foldWith(fa, h, f)
+
+  def fromEither[E, A](ea: Either[E, A]): F[E, A] = ea match {
+    case Left(e)  => raise(e)
+    case Right(a) => pure(a)
+  }
+
+  def fold[E, A, X, R](fa: F[E, A])(h: E => R, f: A => R): F[X, R] =
+    foldWith[E, A, X, R](fa, e => pure(h(e)), a => pure(f(a)))
+
+  def flatMap[E, A, B](fa: F[E, A], f: A => F[E, B]): F[E, B] =
+    foldWith(fa, raise[E, B], f)
+
+  def map[E, A, B](fa: F[E, A])(f: A => B): F[E, B] = flatMap(fa, (a: A) => pure(f(a)))
+
+  def as[E, A, B](fa: F[E, A])(b: => B): F[E, B] = map(fa)(_ => b)
+
+  def void[E, A](fa: F[E, A]): F[E, Unit] = map(fa)(_ => ())
+
+  def flatMapErr[E, A, X](fa: F[E, A], f: E => F[X, A]): F[X, A] =
+    foldWith(fa, f, pure[X, A])
+
+  def mapErr[E, A, X](fa: F[E, A])(f: E => X): F[X, A] =
+    flatMapErr(fa, (e: E) => raise(f(e)))
+
+  def fail[E, A, B](fa: F[E, A])(f: A => E): F[E, B] =
+    flatMap(fa, (a: A) => raise(f(a)))
+
+  def handleWith[E, X, A](fa: F[E, A], h: E => F[X, A]): F[X, A] =
+    foldWith(fa, h, pure[X, A])
+
+  def handle[E, X, A](fa: F[E, A], h: E => A): F[X, A] =
+    handleWith[E, X, A](fa, e => pure(h(e)))
+
+  def errAs[E, A, X](fa: F[E, A])(x: => X) = mapErr(fa)(_ => x)
+
+  def voidErr[E, A](fa: F[E, A]): F[Unit, A] = mapErr(fa)(_ => ())
+
+  def bimap[A, B, C, D](fab: F[A, B])(f: A => C, g: B => D): F[C, D] =
+    foldWith[A, B, C, D](fab, a => raise(f(a)), b => pure(g(b)))
+
+  def swapMap[E, A, X, B](fab: F[E, A])(f: E => B, g: A => X): F[X, B] =
+    foldWith[E, A, X, B](fab, e => pure(f(e)), a => raise(g(a)))
+
+  def swap[E, A](fab: F[E, A]): F[A, E] = foldWith[E, A, A, E](fab, e => pure(e), a => raise(a))
+
+  def left[A, B](a: A): F[A, B] = raise(a)
+
+  def right[A, B](b: B): F[A, B] = pure(b)
+
+  def leftFlatMap[A, B, C, D](lab: F[A, B])(fl: A => F[C, D], fr: B => F[C, D]): F[C, D] = foldWith(lab, fl, fr)
+
+  def rightFlatMap[A, B, C, D](rab: F[A, B])(fl: A => F[C, D], fr: B => F[C, D]): F[C, D] = foldWith(rab, fl, fr)
+
+  def bifunctor: Bifunctor[F] = new Bifunctor[F] {
+    def bimap[A, B, C, D](fab: F[A, B])(f: A => C, g: B => D): F[C, D] = self.bimap(fab)(f, g)
+  }
+
+  override def leftBifunctor: Bifunctor[F] = bifunctor
+
+  override def rightBifunctor: Bifunctor[F] = bifunctor
+}
+
+object TwinMonad extends BindInstanceChain[TwinMonad]
+
+trait Bind[F[_, _]] extends TwinMonad[F] { self =>
```

Comment:
Bad naming. May be confused with mono-Bind (used in scalaz). Should be BiBind

Reply:
I thought it would be so cool since it does binding for symmetric bifunctor monad (gonna write some post\doc about it) and also has "bi" in the name.
Also, cats library doesn't have anything named that, so I've thought it so easy to remember and use.